packages/nextjs/utils/md-parser.ts

```diff
@@ -0,0 +1,137 @@
+};
+
+export const PARSERS = [
+  {
+    marker: "✅",
+    type: "correct",
+    detect: (node: ListItem, marker: string) => getChildText(node).startsWith(marker),
+    clear: (text: string, marker: string) => text.replace(new RegExp(`^${marker}[\s ]?`), ""),
+    apply: (text: string, marker: string) => `${marker} ${text}`,
+  },
+  {
+    marker: "❌",
+    type: "critical",
+    detect: (node: ListItem, marker: string) => getChildText(node).startsWith(marker),
+    clear: (text: string, marker: string) => text.replace(new RegExp(`^${marker}[\s ]?`), ""),
+    apply: (text: string, marker: string) => `${marker} ${text}`,
+  },
+  {
+    marker: "⚠️",
+    type: "acceptable",
+    detect: (node: ListItem, marker: string) => getChildText(node).startsWith(marker),
+    clear: (text: string, marker: string) => text.replace(new RegExp(`^${marker}[\s ]?`), ""),
+    apply: (text: string, marker: string) => `${marker} ${text}`,
+  },
+  {
+    marker: "N/A",
+    type: "not applicable",
+    detect: (node: ListItem) => {
+      // @ts-expect-error
+      return node?.children?.[0].children?.[0].type === "delete";
+    },
+    clear: (_text: string, _marker: string, node: ListItem) => {
+      // @ts-expect-error
+      node.children[0].children[0] = node.children[0].children[0].children[0];
+    },
+    apply: (_text: string, _marker: string, node: ListItem) => {
+      // @ts-expect-error
+      node.children[0].children[0] = {
+        type: "delete",
+        children: [
+          {
+            type: "text",
+            value: getChildText(node),
+          },
+        ],
+      };
+    },
+  },
+] as const;
+
+export type Checkbox = ListItem & {
+  path?: string;
+  conclusion?: (typeof PARSERS)[number]["type"];
+  comment?: string;
+};
+
+const parseCustomMarkers = function (node: Checkbox) {
+  const text = getChildText(node);
+  // check if list item contains one of the known custom markers
+  for (const { marker, type, detect, clear } of PARSERS) {
+    if (detect(node, marker)) {
+      // @ts-expect-error
+      node.children[0].children[0].value = clear(text, marker, node);
+      // record as checked
+      node.checked = true;
+      // record conclusion type
+      node.conclusion = type;
+    }
+  }
+  // if checkbox is not custom
+  if (node.checked === true && node.conclusion === undefined) {
+    // make it default type ("correct")
+    node.conclusion = PARSERS[0].type;
+  }
+};
+
+const renderCustomMarkers = function (node: Checkbox) {
+  if (node.conclusion === undefined) {
+    return;
+  }
+  const text = getChildText(node);
+  node.checked = null;
+  const parser = PARSERS.find(c => c.type === node.conclusion);
+  if (parser) {
+    // @ts-expect-error
+    node.children[0].children[0].value = parser.apply(text, parser.marker, node);
+  }
+};
+
+export const parseMd = async function (text: string) {
+  const tree = await unified().use(remarkParse).use(remarkGfm).parse(text);
+  // console.log(JSON.stringify(tree, null, 2));
```

Comment:
Just saw the `console.log`. This can be deleted if you feel like :D

Reply:
I don't think the commented out log have to be deleted in this project. There are also a bunch of completed TODO comments, which I'm fine with 🙂